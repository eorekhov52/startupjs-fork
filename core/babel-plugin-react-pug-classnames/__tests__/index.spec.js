const pluginTester = require('babel-plugin-tester').default
const plugin = require('../index')
const { name: pluginName } = require('../package.json')

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  pluginOptions: {
    classAttribute: 'styleName'
  },
  babelOptions: {
    plugins: ['@babel/plugin-syntax-jsx']
  },
  tests: {
    "Doesn't transform regular jsx without styleName": /* js */`
      <View style={{ backgroundColor: 'red' }}>
        <Text style={{ color: 'blue' }}>Hello World</Text>
      </View>
    `,
    "Doesn't transform styleName w/o capital letter classes": /* js */`
      <View styleName='card article' style={{ backgroundColor: 'red' }}>
        <Text styleName='content' style={{ color: 'blue' }}>Hello World</Text>
      </View>
    `,
    'Transforms styleName which has a capital class': /* js */`
      <View styleName='Card article' style={{ backgroundColor: 'red' }}>
        <Text styleName='content' style={{ color: 'blue' }}>Hello World</Text>
      </View>
    `,
    'Transforms styleName which has a binary expression': /* js */`
      <View styleName={'Card ' + { article }} style={{ backgroundColor: 'red' }}>
        <Text styleName='content' style={{ color: 'blue' }}>Hello World</Text>
      </View>
    `,
    'Transforms styleName which has a binary expression with multiple static classes': /* js */`
      <View styleName={'Card primary ' + [{ article }, importance]} style={{ backgroundColor: 'red' }}>
        <Text styleName='content' style={{ color: 'blue' }}>Hello World</Text>
      </View>
    `,
    'Transforms binary expressions even when there are no capital classes': /* js */`
      <View styleName={'card primary' + { article }} style={{ backgroundColor: 'red' }}>
        <Text styleName='content' style={{ color: 'blue' }}>Hello World</Text>
      </View>
    `,
    "Doesn't transform array when no capital classes are present": /* js */`
      <View styleName={['card', 'primary']} style={{ backgroundColor: 'red' }}>
        <Text styleName={['content']} style={{ color: 'blue' }}>Hello World</Text>
      </View>
    `,
    'Transform array when capital classes are present': /* js */`
      <View styleName={['Card', 'primary']} style={{ backgroundColor: 'red' }}>
        <Text styleName={['Content']} style={{ color: 'blue' }}>Hello World</Text>
      </View>
    `
  }
})
