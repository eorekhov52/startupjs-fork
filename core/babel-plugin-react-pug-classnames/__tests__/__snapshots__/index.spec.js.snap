// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@startupjs/babel-plugin-react-pug-classnames Doesn't transform array when no capital classes are present: Doesn't transform array when no capital classes are present 1`] = `

<View styleName={['card', 'primary']} style={{ backgroundColor: 'red' }}>
  <Text styleName={['content']} style={{ color: 'blue' }}>Hello World</Text>
</View>

      ↓ ↓ ↓ ↓ ↓ ↓

<View
  styleName={["card", "primary"]}
  style={{
    backgroundColor: "red",
  }}
>
  <Text
    styleName={["content"]}
    style={{
      color: "blue",
    }}
  >
    Hello World
  </Text>
</View>;


`;

exports[`@startupjs/babel-plugin-react-pug-classnames Doesn't transform regular jsx without styleName: Doesn't transform regular jsx without styleName 1`] = `

<View style={{ backgroundColor: 'red' }}>
  <Text style={{ color: 'blue' }}>Hello World</Text>
</View>

      ↓ ↓ ↓ ↓ ↓ ↓

<View
  style={{
    backgroundColor: "red",
  }}
>
  <Text
    style={{
      color: "blue",
    }}
  >
    Hello World
  </Text>
</View>;


`;

exports[`@startupjs/babel-plugin-react-pug-classnames Doesn't transform styleName w/o capital letter classes: Doesn't transform styleName w/o capital letter classes 1`] = `

<View styleName='card article' style={{ backgroundColor: 'red' }}>
  <Text styleName='content' style={{ color: 'blue' }}>Hello World</Text>
</View>

      ↓ ↓ ↓ ↓ ↓ ↓

<View
  styleName="card article"
  style={{
    backgroundColor: "red",
  }}
>
  <Text
    styleName="content"
    style={{
      color: "blue",
    }}
  >
    Hello World
  </Text>
</View>;


`;

exports[`@startupjs/babel-plugin-react-pug-classnames Transform array when capital classes are present: Transform array when capital classes are present 1`] = `

<View styleName={['Card', 'primary']} style={{ backgroundColor: 'red' }}>
  <Text styleName={['Content']} style={{ color: 'blue' }}>Hello World</Text>
</View>

      ↓ ↓ ↓ ↓ ↓ ↓

<View.Card
  styleName={["primary"]}
  style={{
    backgroundColor: "red",
  }}
>
  <Text.Content
    styleName={[]}
    style={{
      color: "blue",
    }}
  >
    Hello World
  </Text.Content>
</View.Card>;


`;

exports[`@startupjs/babel-plugin-react-pug-classnames Transforms binary expressions even when there are no capital classes: Transforms binary expressions even when there are no capital classes 1`] = `

<View styleName={'card primary' + { article }} style={{ backgroundColor: 'red' }}>
  <Text styleName='content' style={{ color: 'blue' }}>Hello World</Text>
</View>

      ↓ ↓ ↓ ↓ ↓ ↓

<View
  styleName={[
    "card",
    "primary",
    {
      article,
    },
  ]}
  style={{
    backgroundColor: "red",
  }}
>
  <Text
    styleName="content"
    style={{
      color: "blue",
    }}
  >
    Hello World
  </Text>
</View>;


`;

exports[`@startupjs/babel-plugin-react-pug-classnames Transforms styleName which has a binary expression with multiple static classes: Transforms styleName which has a binary expression with multiple static classes 1`] = `

<View styleName={'Card primary ' + [{ article }, importance]} style={{ backgroundColor: 'red' }}>
  <Text styleName='content' style={{ color: 'blue' }}>Hello World</Text>
</View>

      ↓ ↓ ↓ ↓ ↓ ↓

<View.Card
  styleName={[
    "primary",
    [
      {
        article,
      },
      importance,
    ],
  ]}
  style={{
    backgroundColor: "red",
  }}
>
  <Text
    styleName="content"
    style={{
      color: "blue",
    }}
  >
    Hello World
  </Text>
</View.Card>;


`;

exports[`@startupjs/babel-plugin-react-pug-classnames Transforms styleName which has a binary expression: Transforms styleName which has a binary expression 1`] = `

<View styleName={'Card ' + { article }} style={{ backgroundColor: 'red' }}>
  <Text styleName='content' style={{ color: 'blue' }}>Hello World</Text>
</View>

      ↓ ↓ ↓ ↓ ↓ ↓

<View.Card
  styleName={[
    {
      article,
    },
  ]}
  style={{
    backgroundColor: "red",
  }}
>
  <Text
    styleName="content"
    style={{
      color: "blue",
    }}
  >
    Hello World
  </Text>
</View.Card>;


`;

exports[`@startupjs/babel-plugin-react-pug-classnames Transforms styleName which has a capital class: Transforms styleName which has a capital class 1`] = `

<View styleName='Card article' style={{ backgroundColor: 'red' }}>
  <Text styleName='content' style={{ color: 'blue' }}>Hello World</Text>
</View>

      ↓ ↓ ↓ ↓ ↓ ↓

<View.Card
  styleName="article"
  style={{
    backgroundColor: "red",
  }}
>
  <Text
    styleName="content"
    style={{
      color: "blue",
    }}
  >
    Hello World
  </Text>
</View.Card>;


`;
